crates/fibre/src/unix.rs

```diff
@@ -134,75 +133,50 @@ extern "C" {
         entry: extern "C" fn(*mut u8, *mut u8),
         entry_arg0: *mut u8,
     );
-    fn wasmtime_fibre_switch(top_of_stack: *mut u8);
+    fn wasmtime_fibre_switch(top_of_stack: *mut u8, payload: u64) -> u64;
     #[allow(dead_code)] // only used in inline assembly for some platforms
     fn wasmtime_fibre_start();
 }
 
-extern "C" fn fiber_start<F, A, B, C>(arg0: *mut u8, top_of_stack: *mut u8)
+extern "C" fn fiber_start<F>(arg0: *mut u8, top_of_stack: *mut u8)
 where
-    F: FnOnce(A, &super::Suspend<A, B, C>) -> C,
+    F: FnOnce((), &super::Suspend) -> (),
 {
     unsafe {
         let inner = Suspend(top_of_stack);
-        let initial = inner.take_resume::<A, B, C>();
-        super::Suspend::<A, B, C>::execute(inner, initial, Box::from_raw(arg0.cast::<F>()))
+        super::Suspend::execute(inner, Box::from_raw(arg0.cast::<F>()))
     }
 }
 
 impl Fiber {
-    pub fn new<F, A, B, C>(stack: &FiberStack, func: F) -> io::Result<Self>
+    pub fn new<F>(stack: &FiberStack, func: F) -> io::Result<Self>
     where
-        F: FnOnce(A, &super::Suspend<A, B, C>) -> C,
+        F: FnOnce((), &super::Suspend) -> (),
```

Comment:
Yes, I didn't make any more changes to this particular piece of code because I want to revamp the type of the continuation function anyway: Using a `FnOnce` means that it's a closure, which then needs to be boxed, causing an allocation. For our use case, we should be able to get away with not needing the closure environment, and can just use a raw function pointer instead. But I will do that in a separate PR.

Reply:
Yes, that resonates with me. I agree with your approach.